org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerWrapper.java

```diff
@@ -471,7 +474,7 @@ public synchronized void stop() {
 		this.lspStreamProvider = null;
 
 		while (!this.connectedDocuments.isEmpty()) {
-			disconnect(this.connectedDocuments.keySet().iterator().next());
+			disconnect(this.connectedDocuments.keySet().iterator().next(), true);
```

Comment:
On the stop all file must be disconnected and should never recall stop or restartStopTimer.

@rubenporras perhaps this disconnect method with stopping parameter could fix your issue https://github.com/eclipse/lsp4e/issues/644 ?